src/isomorphic/devtools/ReactComponentTreeDevtool.js

```diff
@@ -106,6 +106,11 @@ var ReactComponentTreeDevtool = {
   },
 
   purgeUnmountedComponents() {
+    if (ReactComponentTreeDevtool._preventPurging) {
```

Comment:
Not a big fan of these but I’m not sure how else we can test the expected tree during server rendering, if `ReactDebugTool` cleans it up right before the batch. Alternatively we can make those integration tests instead in `ReactPerf-test`, but I like how comprehensive the suite for `ReactComponentDebugTool` is, so I don’t want to replicate the same complexity in `ReactPerf-test`.